app/services/user_creator.rb

```diff
@@ -1,18 +1,19 @@
 # frozen_string_literal: true
 
 class UserCreator
-  def initialize(user_params:, provider:)
+  def initialize(user_params:, provider:, role:)
```

Comment:
IMO, Role feels right as we can in fact pass any role to it.

Reply:
But in that context it will used as a default/fallback role?
A default role is a role, it just indicates that that role will be used by default.
In that context the role coming from `infer_role_from_email` can be named `role` when the other is a fallback role,
However, it's not an issue and I'm okay with both naming schema as long that it's clear (which is).